src/flakeModule.nix

```diff
@@ -4,178 +4,182 @@
   pick,
   winnow,
   harvest,
-}: {
-  inputs,
-  config,
-  options,
-  lib,
-  ...
 }: let
-  inherit
-    (builtins)
-    all
-    concatMap
-    head
-    isList
-    length
-    mapAttrs
-    tail
-    zipAttrsWith
-    ;
-  inherit
-    (lib)
-    getValues
-    mkOption
-    mkOptionType
-    showDefs
-    showOption
-    types
-    literalExpression
-    ;
-
-  # ensures we always end up with a list of lists
-  # if more than one definition
-  mergeHarvesterOption = loc: defs: let
-    list = getValues defs;
-  in
-    if length list == 1
-    then head list
-    else if all isList list
-    then
-      concatMap (l:
-        if isList l
-        then l
-        else [l])
-      list
-    else throw "Cannot merge definitions of `${showOption loc}', as at least one merge candidate is not a list of strings or a list of list of strings. Definition values:${showDefs defs}";
-
-  harvesterType = mkOptionType {
-    name = "harvester";
-    description = "harvest path(s) as a list of strings or a list of lists of strings";
-    descriptionClass = "noun";
-    merge = mergeHarvesterOption;
-    emptyValue = {value = [];};
+  blockTypes = types.BlockTypes "Block Types usable in the Standard flake-parts module";
```

Comment:
Reformatting (via `alejandra`) caused by the introduction of this variable at the top-level scope, which seemed the most appropriate place to avoid collisions in the lower function scope. But open to other suggestions. 

Initially I took the approach of introducing tightly-scoped usages of `with lib.types;` and removing the shadowed variable. I reverted that once I ran into the second issue fixed in f9d73d53a8221f1b79a23853727a78f8774d707b, but I'm thinking it might still be cleaner to go back to that approach.